### PR TITLE
Move pair style eam/cd from USER-MISC to MANYBODY

### PR DESCRIPTION
**Pull request by akohlmey**
## Purpose

This pull request moves the pair style `eam/cd` from the USER-MISC package to the MANYBODY package. The motivation for this move is to simplify dependency handling, as `eam/cd` is derived from the `eam/alloy` pair style. With this move, there are no dependencies for styles in the USER-MISC package left, which in turn simplifies package management for both the conventional and the CMake build system.

## Author(s)

Axel Kohlmeyer (ICTP)

## Backward Compatibility

Apart from the move of sources from one package to another and the corresponding documentation updates, there are no changes and thus it is compatible. Since the pair style `eam/cd` would only be installed, if MANYBODY was also installed, there is no user facing change except that the USER-MISC package may be installed without need.

## Implementation Notes

I vastly prefer this change over hacking CMake to implement the dependency for this one style in the USER-MISC package. In combination with PR #1024 we should have now completely identical handling of dependencies in CMake as in the conventional build system.

## Post Submission Checklist

_Please check the fields below as they are completed_
- [x] The feature or features in this pull request is complete
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [n/a] One or more example input decks are included
- [x] The source code follows the LAMMPS formatting guidelines

### CODE REVIEW

File: src/USER-MISC/README [hunk 1]
```diff
@@ -65,7 +65,6 @@ pair_style buck/mdf, Paolo Raiteri, p.raiteri at curtin.edu.au, 2 Dec 15
 pair_style coul/diel, Axel Kohlmeyer, akohlmey at gmail.com, 1 Dec 11
 pair_style dipole/sf, Mario Orsi, orsimario at gmail.com, 8 Aug 11
 pair_style edip, Luca Ferraro, luca.ferraro at caspur.it, 15 Sep 11
-pair_style eam/cd, Alexander Stukowski, stukowski at mm.tu-darmstadt.de, 7 Nov 09
```
Comment (junghans) [inline src/USER-MISC/README]: Shouldn't the author be mentioned in the eam package now?

Comment (akohlmey) [reply src/USER-MISC/README]: the MANYBODY package doesn't have a README file. this is a special case for USER-MISC. The author (and the corresponding publication) **is** already mentioned in the EAM pair style docs.

**Comment by sjplimp** [comment]
@akohlmey this is fine to move to MANYBODY.  However I request 2 changes, since it
will then be part of standard LAMMPS, not a USER package.  (1) rename the file to
pair_eam_cd.cpp/h and the class to PairEAMCD, (2) indent the source files
the same as others.  The command is already pair eam/cd.  Just want to be consistent.

**Comment by akohlmey** [comment]
@sjplimp it is done

**Pull request merged**